carml-engine/src/test/resources/RmlMapper/test21/ordersNested.rml.ttl

```diff
@@ -0,0 +1,68 @@
+@prefix ex: <http://www.example.com/>.
+@prefix foaf: <http://xmlns.com/foaf/0.1/> .
+@prefix dct: <http://purl.org/dc/terms/> .
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
+@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix rr: <http://www.w3.org/ns/r2rml#> .
+@prefix rml: <http://semweb.mmlab.be/ns/rml#> .
+@prefix ql: <http://semweb.mmlab.be/ns/ql#> .
+@prefix carml: <http://carml.taxonic.com/carml/> .
+@prefix carml-exp: <http://carml.taxonic.com/carml/experimental/> .
+@prefix : <http://www.example.com/mapping/> .
+@prefix ox: <http://www.example.com/orders-vocab/> .
+
+:orders
+  rml:logicalSource [
+    rml:source "test21/ordersNested.json";
+    rml:referenceFormulation ql:JSONPath;
+    rml:iterator "$.orders[*]"
+  ];
+  rr:subjectMap [
+    rr:template "http://www.example.com/orders/{id}";
+    rr:class ox:Order
+  ];
+  rr:predicateObjectMap [
+    rr:predicate dct:date;
+    rr:objectMap [
+      rml:reference "date"
+    ]
+  ];
+  carml-exp:nestedMapping [
+    carml-exp:triplesMap :items;
+
+    # specify 'iteration context'
+    carml-exp:context [
+      carml-exp:key "orderId";
+      carml-exp:valueReference "id"
+    ], [
+      carml-exp:key "orderItems";
+      carml-exp:valueReference "items"
+    ]
+  ]
```

Comment:
The downside of specifying the inclusion in the nested mapping is the tight coupling with the "parent" map, because we have to refer to its fields. So, our example nested map will now only work if it is used from a map that has objects with an "items" field.

My current approach is a bit like calling the sub triples map as a function / subroutine, usable from anywhere. Ofcourse, not sure how often sub triples maps will be used in more than 1 place in practice...

Reply:
Implemented `carml-exp:mergeSuper` based on your comment